common/src/patch.rs

```diff
@@ -0,0 +1,219 @@
+//! Patch-related functions and types.
+use anyhow::{Error, Result};
+use git2::Oid;
+use regex::Regex;
+
+use librad::git::refs::Refs;
+use librad::git::storage::{ReadOnly, ReadOnlyStorage};
+
+use crate::{git, person, project};
+use rad_terminal::components as term;
+
+pub const TAG_PREFIX: &str = "patches/";
+
+#[derive(PartialEq, Eq)]
+pub enum State {
+    Open,
+    Merged,
+}
+
+/// A patch is a change set that a user wants the maintainer to merge into a project's default
+/// branch.
+///
+/// A patch is represented by an annotated tag, prefixed with `radicle-patch/`.
+pub struct Metadata {
+    /// ID of a patch. This is the portion of the tag name following the `radicle-patch/` prefix.
+    pub id: String,
+    /// Peer that the patch originated from
+    pub peer: project::PeerInfo,
+    /// Message attached to the patch. This is the message of the annotated tag.
+    pub message: Option<String>,
+    /// Head commit that the author wants to merge with this patch.
+    pub commit: Oid,
+}
+
+impl Metadata {
+    pub fn name(&self) -> String {
+        str::replace(&self.id, TAG_PREFIX, "")
```

Comment:
`strip_prefix` would be better, since the tag prefix could occur in other parts of the string, and we wouldn't want to remove it there.

Reply:
Actually the id shouldn't have the prefix, so I don't think this function should exist..